src/main.cpp

```diff
@@ -6794,6 +6793,107 @@ ThresholdState VersionBitsTipState(const Consensus::Params& params, Consensus::D
     return VersionBitsState(chainActive.Tip(), params, pos, versionbitscache);
 }
 
+bool LoadMempool(void)
+{
+    FILE* filestr = fopen((GetDataDir() / "mempool.dat").c_str(), "r");
```

Comment:
AFAIK the `fopen` parameters are `restrict`,  and assumed to be there for the lifetime of the file pointer... I don't think this temporary holds those variants and might cause UB?

I could be wrong.